src/python/pants/goal/run_tracker.py

```diff
@@ -191,24 +167,25 @@ def store_stats(self) -> None:
     def has_ended(self) -> bool:
         return self._has_ended
 
-    def end_run(self, outcome: int) -> None:
+    def end_run(self, exit_code: ExitCode) -> None:
         """This pants run is over, so stop tracking it.
 
         Note: If end_run() has been called once, subsequent calls are no-ops.
-
-        :return: PANTS_SUCCEEDED_EXIT_CODE or PANTS_FAILED_EXIT_CODE
         """
 
         if self.has_ended():
             return
 
         self._has_ended = True
 
-        path, duration, self_time, _is_tool = self._main_root_workunit.end()
-        self.cumulative_timings.add_timing(path, duration)
-        self.self_timings.add_timing(path, self_time)
+        if self._run_start_time is None:
+            raise Exception("RunTracker.end_run() called without calling .start()")
+
+        duration = time.time() - self._run_start_time
+
+        self.cumulative_timings.add_timing(label="main", secs=duration)
```

Comment:
We should consider converting this into a single parent (rust side?) workunit that wraps the entire run? It would require some magic in the workunitstore though, probably. Fine to leave it like this for now, but the "cumulative timings" stuff is a stale concept compared to what we have in workunits now.

Reply:
Agreed that this is a stale concept- right now, some toolchain code depends on this `cumulative_timings` entry existing as part of the RunTracker output, so I want to be careful about how we go about removing it.